### PR TITLE
Feature: Add nickname field to the UserData

### PR DESCRIPTION
**Pull request by Kohelbekker**
Feature which add nickname field to the SigUp process and displays it on /profile page if `isNicknamesEnabled()`

### CODE REVIEW

File: src/api/config.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const defaultConfig: Config = {
     showLanding: true,
     sentryEnabled: false,
     captchaLogin: false,
+    nicknames: false,
```
Comment (josadcha) [inline src/api/config.ts:17]: better to name it nicknamesEnabled because it's boolean

File: src/containers/ProfileAuthDetails/index.tsx [hunk 2]
```diff
@@ -117,6 +118,11 @@ class ProfileAuthDetailsComponent extends React.Component<Props, State> {
                         <div className="pg-profile-page__row">
                             <h2>UID: {user.uid}</h2>
                         </div>
+                        {isNicknamesEnabled() ? (
+                            <div className="pg-profile-page__row">
+                                <h2>Nickname: {user.nickname}</h2>
```
Comment (josadcha) [inline src/containers/ProfileAuthDetails/index.tsx:123]: out Nickname to translations

File: src/modules/user/auth/actions.ts [hunk 3]
```diff
@@ -73,6 +73,7 @@ export interface SignInData {
 export interface SignUpFetch {
     type: typeof AUTH_SIGN_UP_FETCH;
     payload: {
+        nickname: string;
```
Comment (josadcha) [inline src/modules/user/auth/actions.ts:76]: nickname should be string | undefined

File: src/screens/SignUpScreen/index.tsx [hunk 4]
```diff
@@ -354,6 +375,7 @@ class SignUp extends React.Component<Props> {
                 case 'recaptcha':
                 case 'geetest':
                     this.props.signUp({
+                        nickname,
                         email,
                         password,
                         captcha_response,
```
Comment (josadcha) [inline src/screens/SignUpScreen/index.tsx:381]: add check here if nickname is enabled

**Review by josadcha — CHANGES_REQUESTED** [review]

File: src/screens/SignUpScreen/index.tsx [hunk 5]
```diff
@@ -344,6 +365,7 @@ class SignUp extends React.Component<Props> {
     private handleSignUp = () => {
         const { configs, i18n, captcha_response } = this.props;
         const {
+            nickname,
             email,
             password,
             refId,
```
Comment (josadcha) [inline src/screens/SignUpScreen/index.tsx:371]: This function looks enormous now, could you please rewrite it
For example
const payload = {
   email,
   password,
   refid: refId,
   data: JSON.stringify({
     language: i18n,
  });

Switch captcha_type
...
payload = { ...payload, captcha response}

if (nicknameEnabled) {
    payload = { ...payload, nickname }
}

**Review by josadcha — CHANGES_REQUESTED** [review]

**Review by josadcha — APPROVED** [review]

**Review by akushniruk — APPROVED** [review]

**Pull request merged**